staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -505,14 +505,12 @@ func (f *DeltaFIFO) Replace(list []interface{}, resourceVersion string) error {
 
 // Resync will send a sync event for each item
 func (f *DeltaFIFO) Resync() error {
-	var keys []string
-	func() {
-		f.lock.RLock()
-		defer f.lock.RUnlock()
-		keys = f.knownObjects.ListKeys()
-	}()
+	f.lock.Lock()
+	defer f.lock.Unlock()
```

Comment:
I surveyed the resync periods across the code base (for both shared informers and one-off controllers) and it looks like we have:

- 15 seconds (PV)
- 30 seconds (a small handful)
- 5 minutes
- 10 minutes
- 12-24 hours (jittered)

I'd like to see them converge toward a larger resync (the 12-24 hours w/jitter), but we will probably need to evaluate why the controllers that use the sub-minute resyncs need such short values.